### PR TITLE
CDRIVER-4309: always use count command for estimatedDocumentCount

### PR DESCRIPTION
**Pull request by jmikola**
https://jira.mongodb.org/browse/CDRIVER-4309

Related to https://github.com/mongodb/mongo-c-driver/pull/982, but I believe this can be merged first.

### CODE REVIEW

**Comment by vector-of-bool** [comment]
Looks good, other than the newly failing `test_estimated_document_count` in `test-mongoc-collection.c`. Just needs to be updated for these changes.

File: src/libmongoc/tests/test-mongoc-collection.c [hunk 1]
```diff
@@ -2502,7 +2501,7 @@ test_estimated_document_count (void)
       collection, tmp_bson ("{'limit': 2, 'skip': 1}"), NULL, &reply, &error);
 
    request = mock_server_receives_msg (
-      server, 0, tmp_bson ("{'aggregate': 'coll', 'limit': 2, 'skip': 1}"));
+      server, 0, tmp_bson ("{'count': 'coll', 'limit': 2, 'skip': 1}"));
```
Comment (jmikola) [inline src/libmongoc/tests/test-mongoc-collection.c:2504]: @vector-of-bool: Thanks for catching this.

**Comment by jmikola** [comment]
Latest build shows [task failures](https://spruce.mongodb.com/version/6275291d57e85a4f7346f29c/tasks) consistent with the base commit (nothing new after restarting the ASAN Ubuntu 18.04 builds), so I think we're good to merge here.

**Review by vector-of-bool — APPROVED** [review]
LGTM

**Pull request merged**